module/zfs/dmu.c

```diff
@@ -2435,6 +2446,9 @@ dmu_write_policy(objset_t *os, dnode_t *dn, int level, int wp, zio_prop_t *zp)
 		nopwrite = (!dedup && (zio_checksum_table[checksum].ci_flags &
 		    ZCHECKSUM_FLAG_NOPWRITE) &&
 		    compress != ZIO_COMPRESS_OFF && zfs_nopwrite_enabled);
+
+		if (os->os_redundant_metadata == ZFS_REDUNDANT_METADATA_ALL)
+			gang_copies++;
```

Comment:
I am not sure whether `ZFS_REDUNDANT_METADATA_MOST` with `zfs_redundant_metadata_most_ditto_level` reduced to 1 makes sense (differs from `ZFS_REDUNDANT_METADATA_ALL`), but since it can be set it seems handling would also make sense here.